flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsMemoryDataManager.java

```diff
@@ -106,14 +111,19 @@ public HsMemoryDataManager(
         ReadWriteLock readWriteLock = new ReentrantReadWriteLock(true);
         this.lock = readWriteLock.writeLock();
 
+        this.subpartitionViewOperationsMap = new ArrayList<>(numSubpartitions);
+        // currently, only HsFullSpillingStrategy supports multiple consumer.
+        final boolean allowMultipleConsumer = spillStrategy instanceof HsFullSpillingStrategy;
```

Comment:
It should be `HsResultPartition`, rather than `HsMemoryDataManager` or `HsSubpartitionMemoryDataManager`, to decide whether multiple consumers should be allowed. This is a property of the entire result partition, not the memory component.